torch/csrc/inductor/aoti_runtime/utils.h

```diff
@@ -230,6 +230,15 @@ inline const ConstantHandle& unwrap_raii_handle_if_needed(
 inline AtenTensorHandle wrap_with_raii_handle_if_needed(
     const ConstantHandle& handle) = delete;
 
+// DANGEROUS.  Do not call unless you explicitly intend to get a reference to a
+// temporary value, which will expire at the end of the current expression.
+// This should only be called in cases where the C-shim API expects an optional
+// input argument (passed by pointer), and a temporary needs to be passed to it.
+template <class T>
+T& temporary_reference(T&& t) {
+  return t;
+}
+
```

Comment:
This is possibly the single most ick-inducing change I have committed in my entire career, but it is guaranteed to work by the language standard until the end of the statement which calls it.  Think of this as effectively the inverse of `std::move()`.

The reason we need this is that it enables us to drop creating should-be-temporary C arrays of tensors in our code (keeping memory unnecessarily alive) when creating arguments for `Optional[List[Tensor]]`, in favor of the following code:

```c++
&temporary_reference(std::vector<AtenTensorHandle>{tensor1, tensor2}.data())
```

If there were any other way I could see around this issue, I would use it, but this is the only one I could see right now.